### DISCUSSION TITLE
Redirecting browsers based on user-agents

### DISCUSSION THREAD

**Discussion by paul-vd** [vercel/next.js#26143] [Help]
Hi,

So I have been having a hard time redirecting users based on their user-agent(s), for instance, if I completely want to drop support for Internet Explorer, I would like to redirect them to a [old-browsers.html](https://vercel.com/old-browser.html) page as vercel does.

Currently, I added a script to my `_document` but this is not optimal, if javascript is disabled

```html
<script
    dangerouslySetInnerHTML={{
      __html: `
      if (/MSIE 10/i.test(navigator.userAgent) || /MSIE 9/i.test(navigator.userAgent) || /rv:11.0/i.test(navigator.userAgent)) {
        window.location.replace("/old-browser.html");
      }
      `,
    }}
  />
```

Optimally this has to be done via the `next.config.js` using the `redirects` option.
I have a hard time understanding the `has` tags and the `source` value to target all `ie` browsers on all links.

As we discussed on discord, @sleeplessbyte proposed a solution to the `has` tags
```
has: [ { type: 'header', key: 'user-agent', value: '.*MSIE.*' }]
```

**Comment by SleeplessByte** [comment]
You might want to combine the headers for IE < 11 and IE = 11, using something like:

> I would try something like source: `*` and value: `.*(MSIE|rv:11\.0).*`
